polynote-frontend/polynote/codec.js

```diff
@@ -107,7 +107,7 @@ export class DataWriter {
     }
 
     ensureBufSize(newSize) {
-        if (this.buffer.byteLength < newSize) {
+        while (this.buffer.byteLength < newSize) {
```

Comment:
Nice catch! But, could it first decide the new size and then reallocate, rather than reallocating continually until it's big enough?

Reply:
Oh sure! I thought there was a reason we needed to go chunk-by-chunk. 
But yeah, it could just figure out how many chunks it needs to add and then add it all in one go?

Like `new ArrayBuffer(this.buffer.byteLength + ceil(newSize/chunkSize) * this.chunkSize);` (with however you do `ceil` in js)

Reply:
Done!